crates/reth/statediff/src/state.rs

```diff
@@ -0,0 +1,126 @@
+use std::collections::HashMap;
+
+use revm_primitives::{alloy_primitives::Address, B256, U256};
+use strata_mpt::{keccak, MptNode, StateAccount, EMPTY_ROOT};
+
+use crate::BatchStateDiff;
+
+/// An (in-memory) representation of the EVM state reconstructed only from [`BatchStateDiff`].
+#[derive(Clone, Default, Debug)]
+pub struct StateReconstructed {
```

Comment:
`ReconstructedState`?

Also the order of items in this module is weird and split up into different parts.

Reply:
done. Also made some effort to put structure into this new crate and move pieces into moreless logical parts.